### ISSUE TITLE
Media: Replace flux selector usage with redux selectors

### ISSUE THREAD

**Issue by sarayourfriend** [Automattic/wp-calypso#43663]
Currently the following `MediaStore` and `MediaListStore` selectors exist. Usages of them should be replaced according to the list below:

<ul><li>MediaStore.get -&gt; Should be replaced with <code>getMediaItem</code></li><li>MediaStore.getAll -&gt; Should be replaced with <code>getMedia</code></li><li>MediaListStore.get -&gt; duplicate of MediaStore.get</li><li>MediaListStore.getAllIds -&gt; gets all media item ids</li><li>MediaListStore.getAll -&gt; calls <code>get</code> with all IDs from <code>getAllIds</code>, unnecessary</li><li>getNextPageQuery -&gt; needs new selector based on the query manager</li><li>hasNextPage -&gt; needs new selector based on the query manager</li><li>isFetchingNextPage -&gt; needs new selector based on the reducer described above</li></ul>

**PRs**
- [ ] `MediaStore.get` and `MediaStore.getAll`
- [x] `MediaListStore.get`, `getAllIds` and `getAll`
- [x] `getNextPageQuery` #44149 
- [x] `isFetchingNextPage` https://github.com/Automattic/wp-calypso/pull/44142
- [x] `hasNextPage`

**Note:** `isFetchingNextPage` will already have been created when the `fetching` reducer is added so this issue is blocked by https://github.com/Automattic/wp-calypso/issues/43660
